Source/WebCore/platform/graphics/cg/PathCG.cpp

```diff
@@ -206,11 +262,30 @@ void PathCG::add(PathRoundedRect roundedRect)
     addBeziersForRoundedRect(roundedRect.roundedRect);
 }
 
+static inline void addToCGContextPath(CGContextRef context, PathRoundedRect segment)
+{
+    // No API to add rounded rects to context.
+    auto path = PathCG::create();
+    path->add(WTFMove(segment));
+    // CGContextAddPath has a bug with existing MoveToPoints in context path.
+    // rdar://118395262
+    auto ctm = CGContextGetCTM(context);
+    auto transformedPath = adoptCF(CGPathCreateCopyByTransformingPath(path->platformPath(), &ctm));
+    CGContextSetCTM(context, CGAffineTransformIdentity);
+    CGContextAddPath(context, transformedPath.get());
+    CGContextSetCTM(context, ctm);
```

Comment:
So this is a workaround for rdar://118395262?

Reply:
Yes.